watch-parties/src/main/java/com/google/sps/servlets/ChatReceiverServlet.java

```diff
@@ -23,18 +23,29 @@
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.sps.data.Chat;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
  
 @WebServlet("/chatstorer")
 public class ChatReceiverServlet extends HttpServlet {
+    UserService userService = UserServiceFactory.getUserService();
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
  
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         String comment = request.getParameter("submitted-comment");
         long timestamp = System.currentTimeMillis();
         String authorID = "User"; //placeholder for now
-
-        // Add Author ID stuff here
+        
+        Query query = new Query("Member").addFilter("user", Query.FilterOperator.EQUAL, userService.getCurrentUser());
+        PreparedQuery results = datastore.prepare(query);
+        for (Entity entity : results.asIterable()) {
```

Comment:
Did you try sean's suggestion of Entity entity = results.asSingleEntity() instead of the for loop.

Also you need to indent on line 47